### PR TITLE
Use module path aliases for imports

### PR DESCRIPTION
**Pull request by katjuell**
### What should this PR do?
Resolves [DEVED-92](https://sourcegraph.atlassian.net/browse/DEVED-92) by adding module path aliases for imports.

Also implements `MarkdownFileWithUrl` as an interface rather than type.

### Why are we making this change?
Nested imports often lead to confusion and a less-than-ideal reading experience. Aliases prevent this by making our code easier to read and reason about.

### What are the acceptance criteria? 
- The app works as it currently does on prod, and the decisions in the PR make sense.

### How should this PR be tested?
- Check out the branch and ensure that everything behaves as expected.

## Pull request process

**Reviewers**:

1. Test functionality using the criteria above.
2. Offer tips for efficiency, feedback on best practices, and possible alternative approaches and things that may not have been considered.
3. For shorter, "quick" PRs, use your best judgement on #​2.
4. Use a collaborative approach and provide resources and/or context where appropriate.
5. Provide screenshots/grabs where appropriate to show findings during review.

**Reviewees**:

1. Prefer incremental and appropriately-scoped changes.
2. Leave a comment on things you want explicit feedback on.
3. Respond clearly to comments and questions.

### CODE REVIEW

File: components/atoms/CreativeCommonsNotice/index.tsx [hunk 1]
```diff
@@ -13,11 +13,19 @@ const CreativeCommonsNotice: React.FunctionComponent = () => (
     <StyledCreativeCommonsNoticeContainer>
         <StyledCreativeCommonsNoticeText>
             This work is licensed under a{' '}
-            <StyledCreativeCommonsLink href={creativeCommonsUrl}>
+            <StyledCreativeCommonsLink 
+                href={creativeCommonsUrl}
+                target="_blank"
```
Comment (katjuell) [inline components/atoms/CreativeCommonsNotice/index.tsx:18]: This isn't related to this PR, but was something I meant to wrap up in the CSS reorg.

File: components/atoms/CreativeCommonsNotice/index.tsx [hunk 2]
```diff
@@ -13,11 +13,19 @@ const CreativeCommonsNotice: React.FunctionComponent = () => (
...
+                rel="noreferrer"
+            >
                 Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License.
             </StyledCreativeCommonsLink>
         </StyledCreativeCommonsNoticeText>
-        <a href={creativeCommonsUrl}>
+        <a 
+            href={creativeCommonsUrl}
+            target="_blank"
```
Comment (katjuell) [inline components/atoms/CreativeCommonsNotice/index.tsx:26]: Same as above.

**Review by ltagliaferri — CHANGES_REQUESTED** [review]
This is cleaner than the relative paths and makes good use of TypeScript's path mapping. Everything working as expected, thanks!

**Review by ltagliaferri — APPROVED** [review]

**Comment by ltagliaferri** [comment]
> This is cleaner than the relative paths and makes good use of TypeScript's path mapping. Everything working as expected, thanks!

Sorry, I pressed "request changes" instead of "approve changes" :doh:

**Pull request merged**